lib/kafkat/command/reassign.rb

```diff
@@ -16,33 +17,139 @@ def run
         opts = Trollop.options do
           opt :brokers, "replica set (broker IDs)", type: :string
           opt :replicas, "number of replicas (count)", type: :integer
+          opt :strategy, "assignment strategy used", type: :string
         end
 
-        broker_ids = opts[:brokers] && opts[:brokers].split(',').map(&:to_i)
+        destination_broker_ids = opts[:brokers] && opts[:brokers].split(',').map(&:to_i)
         replica_count = opts[:replicas]
+        strategy = opts[:strategy]
+        strategy ||= "smart"
 
-        broker_ids ||= zookeeper.get_brokers.values.map(&:id)
+        destination_broker_ids ||= zookeeper.get_brokers.values.map(&:id)
 
-        all_brokers_id = all_brokers.values.map(&:id)
-        broker_ids.each do |id|
-          if !all_brokers_id.include?(id)
+        all_broker_ids = all_brokers.values.map(&:id)
+        destination_broker_ids.each do |id|
+          if !all_broker_ids.include? id
             print "ERROR: Broker #{id} is not currently active.\n"
             exit 1
           end
         end
 
-        # *** This logic is duplicated from Kakfa 0.8.1.1 ***
+        if strategy == "smart"
+          assignments = generate_assignment_smart(topics, destination_broker_ids,
```

Comment:
Let's lay the arguments more clearly: either all on a single line, or one per line. (but not a mix)

See a good presentation:
https://vimeo.com/97329157 minute 22.
 Slides are at http://www.slideshare.net/Kevlin/seven-ineffective-coding-habits-of-many-programmers-45312038/22